packages/webapp/src/components/Animals/RemoveAnimalsModal/index.tsx

```diff
@@ -0,0 +1,185 @@
+  return (
+    <>
+      {/* @ts-ignore */}
+      <Drawer
+        title={t('REMOVE_ANIMALS.REMOVE_ANIMALS')}
+        isOpen={props.isOpen}
+        onClose={props.onClose}
+      >
+        {props.showSuccessMessage ? (
+          <div className={styles.successMessage}>
+            <div>
+              <CheckIcon />
+            </div>
+            <p>{t('REMOVE_ANIMALS.REMOVED_AND_ARCHIVED')}</p>
+          </div>
+        ) : (
+          <form onSubmit={handleSubmit(props.onConfirm)}>
+            {isMobile ? (
+              <fieldset>
+                <legend>{t('REMOVE_ANIMALS.WHY')}</legend>
+                {options.map(({ label, value }) => (
+                  <label key={value} className={styles.mobileOption}>
+                    <input {...register(REASON)} value={value} type="radio" />
+                    <div
+                      className={clsx(
+                        styles.mobileOptionLabel,
+                        value === options[options.length - 1].value
+                          ? styles.mobileOptionLabelRed
+                          : styles.mobileOptionLabelGreen,
+                      )}
+                    >
+                      {label}
+                    </div>
+                  </label>
+                ))}
+              </fieldset>
+            ) : (
+              <Controller
+                name={REASON}
+                control={control}
+                render={({ field }) => (
+                  <ReactSelect
+                    {...field}
+                    // @ts-ignore
+                    label={t('REMOVE_ANIMALS.WHY')}
+                    classNames={{
+                      // @ts-ignore
+                      option: ({ value }) => getReactSelectClassNames(value),
+                      // @ts-ignore
+                      valueContainer: (state) =>
+                        getReactSelectClassNames(state.getValue()[0]?.value),
+                    }}
+                    // @ts-ignore
+                    value={options.find(({ value }) => value === field.value)}
+                    options={options}
+                    onChange={(option: (typeof options)[0]) => field.onChange(option.value)}
+                  />
+                )}
+              />
+            )}
+
+            {!!selectedOption && !isCreatedInErrorSelected && (
+              // @ts-ignore
+              <Input
+                hookFormRegister={register(EXPLANATION)}
+                label={t('REMOVE_ANIMALS.EXPLANATION')}
+                optional
+              />
+            )}
+
+            {!!selectedOption &&
+              (isCreatedInErrorSelected ? (
+                <div className={clsx(styles.removalMessage, styles.textCenter)}>
+                  <WarningIcon />
+                  <p>{t('REMOVE_ANIMALS.WILL_BE_PERMANENTLY_REMOVED')}</p>
+                </div>
+              ) : (
+                <p className={styles.textCenter}>{t('REMOVE_ANIMALS.WILL_BE_ARCHIVED')}</p>
+              ))}
+
```

Comment:
I think the conditions in all of this block could be simplified and be more readable by doing

```
{!!selectedOption && (
   isCreatedInErrorSelected ? (
       <div>...</div>
  ) : (
      <>
         <Input ... />
         <p>...</p>
      </>
   ) 
)}
```

Reply:
Good suggestion!